### DISCUSSION TITLE
Inclusion Error Only When `.terragrunt-cache` Directory is Generated

### DISCUSSION THREAD

**Discussion by giladtele** [gruntwork-io/knowledge-base#782] [Help]
Hi team,

A few days ago, I updated my Terragrunt version to v0.53.0 from v0.48.0. I am applying the same terragrunt modules as the ones I have when I used the older version.   

Immediately after upgrading to the new version, I have started to receive the following error:

```
time=2023-11-07T11:50:53+02:00 level=error msg=Error processing module at 'C:/Users/GiladT/OneDrive - TeleMessage LTD/Desktop/Projects/Work/tm-infrastructure-dev/dev/eks/.terragrunt-cache/qUDVhNPqcKxs9UoJc1_9pxQxKAg/BMLTXhWswTa1pidOH8Db9oRk0RY/eks/terragrunt.hcl'. How this module was found: Terragrunt config file found in a subdirectory of C:/Users/GiladT/OneDrive - TeleMessage LTD/Desktop/Projects/Work/tm-infrastructure-dev. Underlying error: C:/Users/GiladT/OneDrive - TeleMessage LTD/Desktop/Projects/Work/tm-infrastructure-dev/dev/eks/.terragrunt-cache/qUDVhNPqcKxs9UoJc1_9pxQxKAg/BMLTXhWswTa1pidOH8Db9oRk0RY/eks/terragrunt.hcl includes C:/Users/GiladT/OneDrive - TeleMessage LTD/Desktop/Projects/Work/tm-infrastructure-dev/dev/eks/terragrunt.hcl, which itself includes C:/Users/GiladT/OneDrive - TeleMessage LTD/Desktop/Projects/Work/tm-infrastructure-dev/dev/eks/terragrunt.hcl,C:/Users/GiladT/OneDrive - TeleMessage LTD/Desktop/Projects/Work/tm-infrastructure-dev/dev/env.hcl. Only one level of includes is allowed.
time=2023-11-07T11:50:53+02:00 level=error msg=Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```

It's important to clarify two things:
1. This issue only occurs _**AFTER**_ the `.terragrunt-cache` directory is generated by running any form of operation on the modules with `terragrunt-cli`. 
2. This error never occurred, even while taking the cache directory into consideration, in previous versions.
3. This isn't normal behavior, and we shouldn't manually delete cache folders before the next runs. 
4. The error persists across multiple environments and clients (machines running the terraform modules). 
5. This error persists on WSL (Ubuntu 20.04) and Windows 11, but only with the current version of Terragrunt-cli, and not v0.48.0 (at least in my case). 

Attempted solutions:
1. Running the command `terragrunt <OPERATION;APPLY,DESTROY,ETC> --terragrunt-source-update --auto-approve`. It worked, but only with the `--auto-approve` flag. Without it, the cache failed it again, and when applying one module and not all sub-modules under the root terrragrunt modules, as I normally do (aka when running `terragrunt run-all apply --terragrunt-source-update --auto-approve`). 
2. Doing the same but with the flag `-refresh=true`. Failed in all times, no matter which additional flags were provided, with the same original error.
6. Repeating the same, however with the `--terragrunt-exclude-dir '**.terragrunt-cache'` and --terragrunt-exclude-dir '*.terragrunt-cache' flag, using the different arguments I listed; resulting in the same original error.
7. Taking the same course of action, yet this time, using the `terragrunt-use-partial-parse-config-cache` option. Results in the same original error. 

I have thoroughly browsed the known bugs in the versioning changelog and have scanned the internet for similar issues to no avail. Please assist/advise, would greatly appreciate any feedback.

Much obliged,
Gilad

---

<ins datetime="2023-11-07T10:00:40Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110575">Tracked in ticket #110575</a></p>
</ins>

**Comment by denis256** [comment]
Hello,
I was wondering if this is issue happens only on Windows or on other platforms too

**Reply by giladtele** [reply]
Hi,
So far, it happens both on WSL (Ubuntu 20.04) and Windows.
